### DISCUSSION TITLE
Things I'd love to see in Zed

### DISCUSSION THREAD

**Discussion by maxim** [zed-industries/zed#7985] [Show and tell]
I'm a Ruby/Elixir/Go/JS dev who uses Sublime Text, and envies VSCode's Copilot integration, but can't stand the rest of its shifting, fiddly, overloaded UI.

I just tried Zed this weekend for the first time, and it seems amazing. Love out-of-the-box vim mode, Copilot and GPT integration, how it's not slower than Sublime. And unlike VSCode (which I spent months tweaking, trying to switch to it, but still can't), Zed feels immediately comfortable for a Sublime Text user.

Of course, many things are still missing, which is understandable, but I just wanted to list the ones I care about here. I realized I need surprisingly not much. Here they are in order of importance to me:

- [ ] Vertical pane split in Terminal
- [ ] `*.todo` syntax like [PlainTasks](https://packagecontrol.io/packages/PlainTasks) or [Todo+](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-todo-plus) with kb shortcuts to create/mark-done/cancel items
- [x] Run tests with keyboard shortcuts like [RubyTest](https://packagecontrol.io/packages/RubyTest)
- [ ] Have ability to duplicate and move file in the Project sidebar by editing file path like [SideBarEnhancements](https://packagecontrol.io/packages/SideBarEnhancements)
- [ ] Show file's reading time in status bar, like [WordingStatus](https://github.com/evandrocoan/WordingStatus) or [WordCounter](https://marketplace.visualstudio.com/items?itemName=kirozen.wordcounter)
- [ ] Put Terminal to the left of Project dir tree (I really really prefer this placement, but neither VSCode nor Zed will let me do this. For Sublime I use iTerm2 via MacOS split screen.)
- [ ] Set the equivalent of Sublime's `gray_antialias` on a font
- [x] Ability to wrap a block of text to 80char width with a kb shortcut like [Wrap Plus](https://packagecontrol.io/packages/Wrap%20Plus) (albeit AI can do this)

Disclaimer: I don't need everything to be the same, it's a different editor and I'm happy to learn new tricks. Just hope these problems are solved somehow.

Edit: If any of these are already possible, would be thankful for a tip.

**Comment by Tom-Boscher** [comment]
**Vertical pane split in Terminal** is a really needed feature on my side!

**Comment by maxim** [comment]
Ability to wrap a block of text to 80char has been [implemented](https://github.com/zed-industries/zed/pull/17909)! 🎉

**Comment by maxim** [comment]
"Ability to run tests with keyboard shortcuts" has been implemented via task key bindings. However, it would be ideal if there was a way to indicate that Terminal shouldn't steal focus every time you run a task.
